### PR TITLE
feat: DBTP 786 copilot helper version checks

### PR DESCRIPTION
**Pull request by samuele-mattiuzzo**
Jira ticket: https://uktrade.atlassian.net/browse/DBTP-786

### CODE REVIEW

**Comment by samuele-mattiuzzo** [comment]
Still some intermittent tests failing no idea why.
The GH actions part is not in there but can be added later on as I might need some guidance

File: tests/copilot_helper/utils/test_versioning.py [hunk 1]
```diff
@@ -163,3 +187,20 @@ def test_check_copilot_helper_version_skips_when_running_local_version(version_c
...
+
+@patch("click.secho")
+@patch("dbt_copilot_helper.utils.versioning.get_file_app_versions")
+@patch(
+    "dbt_copilot_helper.utils.versioning.running_as_installed_package", new=Mock(return_value=True)
+)
+def test_check_copilot_helper_version_shows_warning_when_different_than_file_spec(
+    get_file_app_versions, secho
+):
+    get_file_app_versions.return_value = (1, 0, 1), (1, 0, 0)
```
Comment (WillGibson) [inline tests/copilot_helper/utils/test_versioning.py:200]:
```suggestion
    get_file_app_versions.return_value = (1, 0, 1), (1, 0, 0)
    
```

**Review by WillGibson — COMMENTED** [review]
I think we also need tests for `pipeline generate` and `copilot make-addons`.

**Comment by samuele-mattiuzzo** [comment]
I'll probably re-do this from scratch to keep it clean

**Pull request closed without merging**